x-pack/packages/ml/url_state/src/url_state.tsx

```diff
@@ -154,7 +161,7 @@ export const UrlStateProvider: FC = ({ children }) => {
       }
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [searchString]
+    [searchStringRef.current]
```

Comment:
❓ I wonder if we need the dependency defined this way 🤔 we rely on reference, hence the dependency list can be empty. 
```suggestion
    []
```

Reply:
You're right of course, we don't need the `ref` as a dependency, updated in 5e4f9a5df1b.